modules/openapi-generator/src/main/resources/dart2/api_client.mustache

```diff
@@ -70,7 +70,7 @@ class ApiClient {
   Future<Response> invokeAPI(
     String path,
     String method,
-    Iterable<QueryParam> queryParams,
+    List<QueryParam> queryParams,
```

Comment:
User of this API should be able to pass `Set` as well, eg. let dart remove duplicated elements and not relay on order of elements in Iterable.

Reply:
I've changed this one based on the call to `_updateParamsForAuth`, this function expect the param to be a list, passing a set would be accepted but would result in runtime error when you call that function

![image](https://user-images.githubusercontent.com/7496871/102719228-a345ff80-42cb-11eb-8afe-f69fe85689c6.png)

```
void main() {
  foo({"a", "b"});
}

void foo(Iterable<String> authNames) {
  bar(authNames);
}

void bar(List<String> authNames) {
}

// Uncaught Error: TypeError: Instance of '_LinkedHashSet<String>': type '_LinkedHashSet<String>' is not a subtype of type 'List<String>'
```

Reply:
There doesn't seem to be any reason for `_updateParamsForAuth` being limited to `List` it's not order dependent either

```dart
  /// Update query and header parameters based on authentication settings.
  /// @param authNames The authentications to apply
  void _updateParamsForAuth(List<String> authNames, List<QueryParam> queryParams, Map<String, String> headerParams) {
    authNames.forEach((authName) {
      Authentication auth = _authentications[authName];
      if (auth == null) throw new ArgumentError("Authentication undefined: " + authName);
      auth.applyToParams(queryParams, headerParams);
    });
  }
```

Set seems to be even more correct options as some frameworks (I remember one PHP based doing it) will throw error if you pass the same url param twice.

Reply:
Well, what you said make sense, I'm gonna try to change the signature of `_updateParamsForAuth` and `applyToParams` to receive a `Iterable` instead of a `List` then

Reply:
I forgot to thank you for the review, thanks for that :heart: 

It turns out that I can't change the signature of `applyToParams` to a `Iterable`: in `ApiKeyAuth` there's a call to `.add`, and Iterables don't have that method :/

We could change the method and add a `.toList()` call:
```dart
  void _updateParamsForAuth(
    List<String> authNames,
    Iterable<QueryParam> queryParams,
    Map<String, String> headerParams,
  ) {
    for(final authName in authNames) {
      final auth = _authentications[authName];
      if (auth == null) {
        throw ArgumentError('Authentication undefined: $authName');
      }
      auth.applyToParams(queryParams.toList(), headerParams);
    }
  }
```

But I'm pretty sure that this would make the `applyToParams` implementers `.add` calls pointless.. they rely on mutability, and `.toList` would create a copy of the original list

Reply:
Does that mean it currently throws at runtime if we pass Set in place of the existing iterable, which at the end is expected to be a List?

Reply:
so basically for now it has to be specific implementation Set or List but not parent class `Iterable` because it doesn't have `.add` method :clap:. 

sure, leave it as list then

Reply:
>We could change the method and add a .toList() call:

My personal choice would be to pass iterable to allows sets and any other lists and sets, but it's also outside of scope of your PR, so I'm leaving the decision with you.

Reply:
I'm gonna keep it as a List, ideally we would use iterable and not rely on mutability (as it may affect code outside of this generated lib).

I'm gonna try to open another PR to make it accept a iterable and also rely on pure functions instead of mutating the list

Reply:
Is this external API? If so, this is a breaking change, 5.0.0 was just released. But it also fixes a bug so I am not sure what the process is here. @agilob You gonna fix this before the next release? :)